device/src/main/java/ru/startandroid/device/dialog/DialogHelper.kt

```diff
@@ -0,0 +1,32 @@
+package ru.startandroid.device.dialog
+
+import android.app.Fragment
+import android.content.Intent
+import javax.inject.Inject
+
+class DialogHelper @Inject constructor() {
+
+    private val configs: MutableMap<Int, DialogConfig> = mutableMapOf()
+
+    fun registerDialogConfig(dialogCode: Int, dialogConfig: DialogConfig) {
+        configs[dialogCode] = dialogConfig
+    }
+
+    fun showDialog(dialogCode: Int, targetFragment: Fragment) {
+        DialogFragment.newInstance(dialogCode, configs[dialogCode], targetFragment)
+                .show(targetFragment.activity.fragmentManager, "dialog")
+    }
+
+    fun handleResult(dialogCode: Int, resultCode: Int, data: Intent?) {
+        configs[dialogCode]?.run {
+            when (resultCode) {
+                DialogFragment.RESULT_POSITIVE -> positiveAction
```

Comment:
Can't get what are positiveAction, negativeAction and neutralAction... This are functions, and as a result of when block we launch them with data parameter, but where are they themselves?

Reply:
They are in configs[dialogCode].  

```
configs[dialogCode]?.run {
// here we can see all members of config we get from map configs[dialogCode]  
}
```

It's a Kotlin )